app/commands/solution/search_community_solutions.rb

```diff
@@ -9,18 +9,22 @@ def self.default_per
       DEFAULT_PER
     end
 
-    def initialize(exercise, page: nil, per: nil)
+    def initialize(exercise, page: nil, per: nil, criteria: nil)
       @exercise = exercise
       @page = page.present? && page.to_i.positive? ? page.to_i : DEFAULT_PAGE # rubocop:disable Style/ConditionalAssignment
       @per = per.present? && per.to_i.positive? ? per.to_i : self.class.default_per # rubocop:disable Style/ConditionalAssignment
+      @criteria = criteria
     end
 
     def call
       @solutions = exercise.solutions.published
+
+      @solutions = solutions.joins(:user).where("users.handle LIKE ?", "%#{criteria}%") if @criteria.present?
```

Comment:
@iHiD Just temporary for the tests to pass. You might want to change this to have better performance.